manager/state/raft/membership/cluster.go

```diff
@@ -137,18 +146,36 @@ func (c *Cluster) ClearMember(id uint64) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	if c.members[id] != nil {
-		conn := c.members[id].Conn
-		if conn != nil {
-			_ = conn.Close()
+	return c.clearMember(id)
+}
+
+func (c *Cluster) clearMember(id uint64) error {
+	m, ok := c.members[id]
+	if ok {
+		if m.Conn != nil {
+			m.Conn.Close()
 		}
+		m.hb.Stop()
 		delete(c.members, id)
 	}
-
 	c.broadcastUpdate()
 	return nil
 }
 
+func (c *Cluster) newHeartbeat(m *Member) *heartbeat.Heartbeat {
+	return heartbeat.New(c.heartbeatInterval, func() {
+		c.mu.Lock()
```

Comment:
I guess it's racy because the callback could be blocking on acquiring this lock, during a call to `ReportActive`? So `ReportActive sets `active` to true, but then this clears it immediately.

Maybe there should be a timestamp of the most recent activity stored on the member, and this callback should not set `active` to false if that timestamp is too recent.